hera_pspec/pspecdata.py

```diff
@@ -620,18 +655,22 @@ def cross_covar_model(self, key1, key2, model='empirical',
         assert isinstance(key1, tuple), "key1 must be fed as a tuple"
         assert isinstance(key2, tuple), "key2 must be fed as a tuple"
         assert isinstance(model, (str, np.str)), "model must be a string"
-        assert model in ['empirical'], "didn't recognize model {}".format(model)
-
-        # parse key
-        dset, bl = self.parse_blkey(key1)
-        key1 = (dset,) + (bl,)
-        dset, bl = self.parse_blkey(key2)
-        key2 = (dset,) + (bl,)
-
+        assert model in ['empirical', 'dsets'], "didn't recognize model {}".format(model)
         if model == 'empirical':
-            covar = utils.cov(self.x(key1), self.w(key1),
-                              self.x(key2), self.w(key2),
+            # parse key
+            dset, bl = self.parse_blkey(key1)
+            key1 = (dset,) + (bl,)
+            dset, bl = self.parse_blkey(key2)
+            key2 = (dset,) + (bl,)
+
+            covar = utils.cov(self.x(key1, filter_extension=True), self.w(key1, filter_extension=True),
+                              self.x(key2, filter_extension=True), self.w(key2, filter_extension=True),
                               conj_1=conj_1, conj_2=conj_2)
+        elif model == 'dsets':
+            covar = np.zeros((self.spw_Nfreqs,
+                              self.spw_Nfreqs),
+                              dtype=complex)
+        #for dsets, we assume no baseline-baseline covariances.
         return covar
 
     def I(self, key):
```

Comment:
Right now, I does not extend with non-trivial filter_extensions and will break (thanks @saurabh-astro  for finding this).

Reply:
I've fixed this.